/*****************************************************************************
 * PokerTH - The open source texas holdem engine                             *
 * Copyright (C) 2006-2013 Felix Hammer, Florian Thauer, Lothar May          *
 *                                                                           *
 * This program is free software: you can redistribute it and/or modify      *
 * it under the terms of the GNU Affero General Public License as            *
 * published by the Free Software Foundation, either version 3 of the        *
 * License, or (at your option) any later version.                           *
 *                                                                           *
 * This program is distributed in the hope that it will be useful,           *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of            *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the             *
 * GNU Affero General Public License for more details.                       *
 *                                                                           *
 * You should have received a copy of the GNU Affero General Public License  *
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.     *
 *                                                                           *
 *                                                                           *
 * Additional permission under GNU AGPL version 3 section 7                  *
 *                                                                           *
 * If you modify this program, or any covered work, by linking or            *
 * combining it with the OpenSSL project's OpenSSL library (or a             *
 * modified version of that library), containing parts covered by the        *
 * terms of the OpenSSL or SSLeay licenses, the authors of PokerTH           *
 * (Felix Hammer, Florian Thauer, Lothar May) grant you additional           *
 * permission to convey the resulting work.                                  *
 * Corresponding Source for a non-source form of such a combination          *
 * shall include the source code for the parts of OpenSSL used as well       *
 * as that of the covered work.                                              *
 *****************************************************************************/

option java_package = "de.chatcleaner.protocol";
option java_outer_classname = "ProtoBuf";
option optimize_for = LITE_RUNTIME;

message CleanerInitMessage {
	required	uint32			requestedVersion = 1;
	required	string			clientSecret = 2;
}

message CleanerInitAckMessage {
	required	uint32			serverVersion = 1;
	required	string			serverSecret = 2;
}

enum CleanerChatType {
	cleanerChatTypeLobby = 0;
	cleanerChatTypeGame = 1;
}

message CleanerChatRequestMessage {
	required	uint32			requestId = 1;
	required	CleanerChatType	cleanerChatType = 2;
	optional	uint32			gameId = 3 [default = 0];
	required	uint32			playerId = 4;
	required	string			playerName = 5;
	required	string			chatMessage = 6;
}

message CleanerChatReplyMessage {
	required	uint32			requestId = 1;
	required	CleanerChatType	cleanerChatType = 2;
	optional	uint32			gameId = 3 [default = 0];
	required	uint32			playerId = 4;
	enum CleanerActionType {
		cleanerActionNone = 0;
		cleanerActionWarning = 1;
		cleanerActionKick = 2;
		cleanerActionBan = 3;
		cleanerActionMute = 4;
	}
	required	CleanerActionType	cleanerActionType = 5;
	optional	string			cleanerText = 6 [default = ""];
}

// The main message type (it is prefixed by 4 bytes length of the message).

message ChatCleanerMessage {
	enum ChatCleanerMessageType {
		Type_CleanerInitMessage = 1;
		Type_CleanerInitAckMessage = 2;
		Type_CleanerChatRequestMessage = 3;
		Type_CleanerChatReplyMessage = 4;
	}
	required	ChatCleanerMessageType			messageType = 1;

	optional	CleanerInitMessage				cleanerInitMessage = 2;
	optional	CleanerInitAckMessage			cleanerInitAckMessage = 3;
	optional	CleanerChatRequestMessage		cleanerChatRequestMessage = 4;
	optional	CleanerChatReplyMessage			cleanerChatReplyMessage = 5;
}
